lusidtools/cocoon/utilities.py

```diff
@@ -816,9 +817,28 @@ def handle_nested_default_and_column_mapping(
 
         # If the value of the mapping is a dictionary
         if isinstance(value, dict):
+            nested_key = key
+
+            # If an 'instrument_type' is specified within mapping, expand nested definition mapping
+            if "instrument_type" in mapping.keys():
+                mapping_obj_df_key = mapping["instrument_type"]
+                mapping_object = data_frame.iloc[0][mapping_obj_df_key]
+
+                # Check provided instrument_type is a valid complex instrument value
+                valid_instrument_types = (
+                    models.LusidInstrument.discriminator_value_class_map
+                )
+                if mapping_object not in valid_instrument_types:
+                    raise TypeError(
+                        f"""The specified instrument type {mapping_object} is invalid.
+                        Please set instrument type to one of: {valid_instrument_types.keys()}"""
+                    )
+                else:
+                    for key, value in mapping[nested_key].items():
+                        mapping_updated[key] = value
```

Comment:
Even if the mapping updated how would this get used later on? There is no field called `dom_ccy` as per your test the mapping would need to be `definition.dom_ccy`.